internals/daemon/daemon.go

```diff
@@ -166,18 +166,15 @@ func (c *Command) canAccess(r *http.Request, user *UserState) accessResult {
 
 	// isUser means we have a UID for the request
 	isUser := false
-	pid, uid, socket, err := ucrednetGet(r.RemoteAddr)
+	ucred, err := ucrednetGet(r.RemoteAddr)
 	if err == nil {
 		isUser = true
 	} else if err != errNoID {
 		logger.Noticef("Cannot parse UID from remote address %q: %s", r.RemoteAddr, err)
 		return accessForbidden
 	}
 
-	isUntrusted := (socket == c.d.untrustedSocketPath)
-
-	_ = pid
-	_ = uid
+	isUntrusted := (ucred != nil && ucred.Socket == c.d.untrustedSocketPath)
```

Comment:
I think we can just remove the `ucred != nil` part, as functions that return error like this are almost always guaranteed to never return a `nil` pointer if `err == nil` -- this is true of `ucrednetGet` as well.

Reply:
Oh, oops, I missed the fact that we're only handling the `err != errNoID` case above, but if `err == errNoID` it could still end up here with a nil `ucred`. Yeah, I think what you have now is fine.